fox_news_scraper/extract_fn.py

```diff
@@ -0,0 +1,68 @@
+"""A file to extract information from various Fox News RSS feeds."""
+
+import feedparser
+import requests
+from bs4 import BeautifulSoup
+
+
+def fetch_rss_feed(feed_url: str) -> feedparser.FeedParserDict:
+    """Fetches and parsers the RSS feed from the provided URL."""
+
+    feed = feedparser.parse(feed_url)
+    return feed
+
+
+def get_article_content(article_url: str) -> str:
+    """Scrapes the full content of an article from a given URL."""
+
+    try:
+        response = requests.get(article_url, timeout=10)
+        soup = BeautifulSoup(response.content, "html.parser")
+
+        article_body = soup.find("div", class_="article-body")
+        if article_body:
+            return article_body.get_text(separator=" ", strip=True)
+
+        return "Full content not found or unable to parse."
+
+    except Exception as e:
+        return f"Failed to fetch full content from {article_url}: {e}"
+
+
+def parse_feed_entries(feed: feedparser.FeedParserDict) -> list[dict]:
+    """Parses the entries of a feed and extracts relevant fields."""
+
+    entries = []
+    for entry in feed.entries:
+        content = get_article_content(entry.link)
```

Comment:
It would be good idea to use asynchronous calls here since there could be alot of entries. See grequest library, it works in a Lambda too and isn't too complicated to use.

Here is an example from a previous project of using it:
```
import grequests

def get_plant_data(plant_ids: list[int]):
    """Sends simultaneous calls to the API."""

    async_list = []
    for i in plant_ids:
        url = f"{BASE_URL}{i}"
        action_item = grequests.get(url, timeout=TIMEOUT)
        async_list.append(action_item)

    responses = grequests.map(async_list)

    results = []
    for response in responses:
        result = handle_response(response)
        results.append(result)

    return results
```

Reply:
Great idea, I'll implement that!